rust/src/http2/http2.rs

```diff
@@ -160,6 +168,46 @@ impl HTTP2Transaction {
         }
     }
 
+    #[cfg(not(feature = "decompression"))]
+    fn handle_headers(&mut self, _blocks: &Vec<parser::HTTP2FrameHeaderBlock>, _dir: u8) {}
+
+    #[cfg(feature = "decompression")]
+    fn handle_headers(&mut self, blocks: &Vec<parser::HTTP2FrameHeaderBlock>, dir: u8) {
+        for i in 0..blocks.len() {
+            if blocks[i].name == "content-encoding".as_bytes().to_vec() {
+                self.decoder.http2_encoding_fromvec(&blocks[i].value, dir);
+            }
+        }
+    }
+
+    // dir may be unused without the decompression feature
+    fn decompress<'a>(
+        &'a mut self, input: &'a [u8], _dir: u8, sfcm: &'static SuricataFileContext, over: bool,
+        files: &mut FileContainer, flags: u16,
+    ) -> io::Result<()> {
+        #[cfg(feature = "decompression")]
+        let mut output = Vec::with_capacity(decompression::HTTP2_DECOMPRESSION_CHUNK_SIZE);
+        #[cfg(feature = "decompression")]
+        let decompressed = self.decoder.decompress(input, &mut output, _dir)?;
+        #[cfg(not(feature = "decompression"))]
```

Comment:
instead of these conditions I think its also possible to define a method twice. One would support the decoder, other would just pass through data?

SMB has an example
```rust
#[cfg(not(feature = "debug"))]
fn debug_add_progress(_js: &mut JsonBuilder, _tx: &SMBTransaction) -> Result<(), JsonError> { Ok(()) }

#[cfg(feature = "debug")]
fn debug_add_progress(jsb: &mut JsonBuilder, tx: &SMBTransaction) -> Result<(), JsonError> {
    jsb.set_bool("request_done", tx.request_done)?;
    jsb.set_bool("response_done", tx.response_done)?;
    Ok(())
}
```

Reply:
Agreed.

Reply:
> instead of these conditions I think its also possible to define a method twice. One would support the decoder, other would just pass through data?

Indeed, it is possible to define a method twice as is done for `handle_headers`

But I do not see how it can be implemented in this case because of `output` lifetime which needs to reach `self.ft.new_chunk`

Ie this patch fails compilation with decompression enabled with 
```
error[E0515]: cannot return value referencing local variable `output`
   --> src/http2/http2.rs:191:16
    |
191 |         return self.decoder.decompress(input, &mut output, _dir);
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-----------^^^^^^^
    |                |                              |
    |                |                              `output` is borrowed here
    |                returns a value referencing data owned by the current function


```